include/zephyr/spinlock.h

```diff
@@ -257,10 +322,30 @@ static ALWAYS_INLINE void k_spin_unlock(struct k_spinlock *l,
 	 * Zephyr framework for that.
 	 */
 	atomic_clear(&l->locked);
-#endif
+#endif /* CONFIG_TICKET_SPINLOCKS */
+#endif /* CONFIG_SMP */
 	arch_irq_unlock(key.key);
 }
 
+/**
+ * @brief Checks if spinlock is held by some CPU, including the local CPU
+ *
+ * @param l A pointer to the spinlock to release
+ * @retval true - if spinlock is held by some CPU; false - otherwise
+ */
+#ifdef CONFIG_SMP
+static ALWAYS_INLINE bool k_spin_is_locked(struct k_spinlock *l)
```

Comment:
I'm a bit worried about this new API.

In general it's not legal to ask "is this spinlock is locked?", because it can be locked / unlocked before you can use the answer. And I can't imagine the context on SMP system where it would work without issues.

As we introduce this API just for tests - probably we should guard it with something like `CONFIG_TEST` - so the API can be used in tests only? And we should add note in the function description as well.

Reply:
Hi @evgeniy-paltsev ,

I agree, imho it only makes sense for white-box testing purpose so wrapping it into CONFIG_TEST seems like a good idea. I will prepare a separate MR for this API